PriceCalculatorKata/Product.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace PriceCalculatorKata
+{
+    public class Product
+    {
+        public string Name { get; set; }
+        public int UPC { get; }
+
+        private float _basePrice;
+        public float BasePrice
+        {
+            get
+            {
+                return _basePrice;
+            } set {
+                if(value > 0)
+                {
+                    _basePrice = (float)Math.Round(value, 2);
+                }   
+            }
+        }
+
+        public float PriceAfterTax
+        {
+            get
+            {
+                return (float)Math.Round(BasePrice + (TaxPercentage / 100.0f) * BasePrice,2);
+            }
+        }
+        public int TaxPercentage { get; set; }
+
+        public Product(int UPC)
+        {
+            this.UPC = UPC;
+            this.TaxPercentage = 20;
```

Comment:
maybe it's better to have the default Tax as a constant variable in the product class or  it's a separated constants class.